x/airdrop/keeper/grpc_query.go

```diff
@@ -36,9 +36,17 @@ func (q querier) AirdropAccount(
 	req *types.AirdropAccountRequest,
 ) (*types.AirdropAccountResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	airdropAccount, err := q.GetAirdropAccount(ctx, req.Address)
-	if err != nil {
-		return nil, err
+	airdropAccount, err := q.GetAirdropAccount(ctx, req.Address, types.StateCreated)
+	if err == nil {
+		return &types.AirdropAccountResponse{AirdropAccount: airdropAccount}, nil
 	}
-	return &types.AirdropAccountResponse{AirdropAccount: airdropAccount}, nil
+	airdropAccount, err = q.GetAirdropAccount(ctx, req.Address, types.StateUnclaimed)
+	if err == nil {
+		return &types.AirdropAccountResponse{AirdropAccount: airdropAccount}, nil
+	}
+	airdropAccount, err = q.GetAirdropAccount(ctx, req.Address, types.StateClaimed)
+	if err == nil {
+		return &types.AirdropAccountResponse{AirdropAccount: airdropAccount}, nil
+	}
+	return nil, err
```

Comment:
This is super ugly, but I get that it's because the state is included in the key. I don't see a better way of doing this

Reply:
True, the trade of being able to iterate over a specific state is that you have to query them separately. In this case I chose the grpc query that will barely ever get used to be the less efficient KV store interaction over the end blocker iterators. I could require the state as one of the query params to cleanup the code here, but I kind of like how it functions currently.

Reply:
Yeah I think it's best for the client & performance